src/history/historyTracker.ts

```diff
@@ -93,6 +93,50 @@ class HistoryStep {
     this.cursorStart = init.cursorStart || undefined;
     this.marks     = init.marks || [];
   }
+
+  /**
+   * merge collapses individual character changes into larger blocks of changes
+   */
+  public merge(): void {
+    if (this.changes.length < 2) {
+      return;
+    }
+    // create new changes list
+    var merged: DocumentChange[] = [];
+    // iterate through the changes, saving the first outside the loop
+    var iter = (this.changes[Symbol.iterator]());
+    var prev = iter.next().value;
+    for (const change of iter) {
+      if (prev.text.length === 0) {
```

Comment:
This loop is basically an `Array.reduce()`, but we can't use that directly because it can output more than one DocumentChange. e.g. changes in different parts of the document or a del+add (see below). reduce passes two arguments to the function: `previous` and `current`.

Here we have `prev` and `change` instead.